### ISSUE TITLE
`resolution` keyword to Scene.load not having desired effect

### ISSUE THREAD

**Issue by mraspaud** [pytroll/satpy#116]
When loading a composite with the `resolution` keyword set, the resolution chosen is not always respected. This behaviour was shown by running on modis data in both python 2.7 and 3.6. In 2.7, the data loaded was 1000m in 3.6 it was the maximum resolution available.

Here is the test script:
```python
    scn = Scene(platform_name="Aqua", sensor="modis",
                base_dir="/home/a001673/data/satellite",
                reader='hdfeos_l1b',
                start_time=datetime(2016, 8, 5, 0, 0),
                end_time=datetime(2016, 8, 15, 0, 0))

    scn.load(['true_color'], resolution=1000)
```

**Comment by davidh-ssec** [comment]
@mraspaud Both environments were running the newest version of satpy? This was supposed to be fixed a while ago so that the highest resolution available is provided. Are the individual bands all provided at the highest resolution for those bands? If this isn't consistent between machines even when using the same version of python I would guess that it is a dict key sorting problem. Meaning we aren't sorting some `dict.keys()` when we should be.

**Comment by davidh-ssec** [comment]
Fixed by #138.
